src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -140,40 +135,61 @@ private OverdueFineCalculatorService(Clients clients) {
       return completedFuture(succeeded(params));
     }
-    return feeFineRepository.getOverdueFine(params.feeFineOwner.getId())
+    return repos.feeFineRepository.getOverdueFine(params.feeFineOwner.getId())
       .thenApply(ResultBinding.mapResult(params::withFeeFine))
       .thenCompose(r -> r.after(updatedParams -> {
         if (updatedParams.feeFine == null) {
           FeeFine feeFine = new FeeFine(UUID.randomUUID().toString(),
             updatedParams.feeFineOwner.getId(), FeeFine.OVERDUE_FINE_TYPE);
-          return feeFineRepository.create(feeFine)
+          return repos.feeFineRepository.create(feeFine)
             .thenApply(ResultBinding.mapResult(params::withFeeFine));
         }
         return completedFuture(succeeded(updatedParams));
       }));
   }
 
-  private CompletableFuture<Result<Account>> createFeeFineRecord(Loan loan, Double fineAmount) {
+  private CompletableFuture<Result<CalculationParameters>> lookupLoggedInUser(
+    CalculationParameters params, String loggedInUserId) {
+    return repos.userRepository.getUser(loggedInUserId)
+      .thenApply(ResultBinding.mapResult(params::withLoggedInUser));
+  }
+
+  private CompletableFuture<Result<Account>> createFeeFineRecord(Loan loan, Double fineAmount,
+    String loggedInUserId) {
+
     if (fineAmount <= 0) {
       return failure();
     }
 
     return CompletableFuture.completedFuture(succeeded(
-      new CalculationParameters(loan, null, null, null)))
+      new CalculationParameters(loan, null, null, null, null)))
       .thenCompose(r -> r.after(this::lookupItemRelatedRecords))
       .thenCompose(r -> r.after(this::lookupFeeFineOwner))
       .thenCompose(r -> r.after(this::lookupFeeFine))
+      .thenCompose(r -> r.after(params -> this.lookupLoggedInUser(params, loggedInUserId)))
       .thenCompose(r -> r.after(params -> {
         if (params.item == null || params.feeFineOwner == null || params.feeFine == null) {
           return failure();
         }
 
-        AccountStorageRepresentation account = new AccountStorageRepresentation(params.loan,
-          params.item, params.feeFineOwner, params.feeFine, fineAmount);
-        return accountRepository.create(account);
+        AccountStorageRepresentation accountRepresentation =
+          new AccountStorageRepresentation(params.loan, params.item, params.feeFineOwner,
+            params.feeFine, fineAmount);
+
+        return repos.accountRepository.create(accountRepresentation)
+          .thenCompose(rac -> rac.after(account -> this.createFeeFineAction(account, params)
+            .thenApply(rfa -> rfa.map(feeFineAction -> account))));
       }));
   }
 
+  private CompletableFuture<Result<FeeFineAction>> createFeeFineAction(
+    Account account, CalculationParameters params) {
+
+    return repos.feeFineActionRepository.create(new FeeFineActionStorageRepresentation(account,
+      params.feeFineOwner, params.feeFine, account.getAmount(), account.getAmount(),
```

Comment:
Amount means "by how much this action is changing the balance". They are equal in this case because charging a fine is the first action. Later the balance may be changed by other actions - payment, waive, refund etc. Fee fine actions represent history of such events, for example:
```
Action   |  Amount  |   Balance
-------------------------------
Charge         5          5
Waive          2          3
Pay            3          0
```